subsys/net/lib/CMakeLists.txt

```diff
@@ -3,7 +3,8 @@ add_subdirectory_if_kconfig(lwm2m)
 add_subdirectory_if_kconfig(sntp)
 add_subdirectory_ifdef(CONFIG_DNS_RESOLVER      dns)
 add_subdirectory_ifdef(CONFIG_MQTT_LIB          mqtt)
-add_subdirectory_ifdef(CONFIG_NET_APP_SETTINGS  app)
+add_subdirectory_ifdef(CONFIG_NET_APP           app)
+add_subdirectory_ifdef(CONFIG_NET_APP_SETTINGS  config)
```

Comment:
Do we want it as a library or as a service? It could do stuff once like it does right now as a function call, but it could also do something a bit more advanced such as running dhcpv4 on relevant interfaces, or try some reconnection logic (this will be relevant for wifi for instance). Not that I am advocating to have something as advanced as ConnMan but something better than just 1 function call that user needs to call by himself.

Reply:
This is now a service but we can change this as needed. I think we do not yet know how the device setup should be done so patches are welcome. I suggest that we merge this like it is now as this should not cause any regression how it is done atm.

Reply:
> this should not cause any regression how it is done atm.

Yes, that was the aim - to just split files from one dir into two, but keeping the rest of things as is. And in my few, the config library is an API (with just one call net_app_init(), to be renamed), as well as once-off, bootstrap "service" to call it on boot. There's currently no "run persistently" part in it. If we find a need to make it such, we can do that later of course.